core/templates/components/skill-selector/select-skill-modal.component.ts

```diff
@@ -41,7 +41,8 @@ export class SelectSkillModalComponent extends ConfirmOrCancelModal {
   }
 
   confirm(): void {
-    let totalSkills = [];
+    let totalSkills:
+    (ShortSkillSummary | SkillSummaryBackendDict | SkillSummary)[] = [];
```

Comment:
Why this has some many different types.

Reply:
![image](https://user-images.githubusercontent.com/37558983/159710194-ce4c089a-f785-4779-9655-9b432fcc8be4.png)

Please allow me to use this screenshot to explain why totalSkills has items of different types precisely 3.

1. In the body of the first if statement (line 47) , `totalSkills` is assigned an array created from the `skillSumaries` property which is of type `SkillSumary[]` with the spread operator(...)  and therefore contains items of type `SkillSummary`.

2. In the body of the second if statement (50), items of the `untriagedSkillSummaries` property with type ShortSkillSummary  are pushed to  `totalSkills`. This means that totalSkills needs to also accept items of type `ShortSkillSummary`.

3. In the body of the last `if statement` (line 55),  items of the `categorizedSkills[topic][subtopic]` property with type ShortSkillSummary  are pushed to  `totalSkills`. This means that totalSkills needs to also accept items of type `ShortSkillSummary`.

Reply:
Okay sir, I've figured out the redundant type in the `totalSkills` array. 

Actually in my first point, the `skillSummary` array contains items of type `SkillSummaryBackendDict` not `SkillSummary`

So in the end `totalSkills` is an array of two types of items: `SkillSummaryBackendDict` and `ShortSkillSummary`.
Please @vojtechjelinek  check whether my analysis is okay so I can make the changes and push.
Thanks

Reply:
Great analysis, I think we should try to reduce the number of types thought. Is it possible to somehow transform the values so that they are of one type? If not then I think two types that would make a bit of sense are `ShortSkillSummary` and `SkillSummary`.